modules/tinymce/src/plugins/quickbars/test/ts/browser/KeyboardShortcutContextualMenuTest.ts

```diff
@@ -0,0 +1,33 @@
+import { FocusTools, Keys } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { SugarDocument } from '@ephox/sugar';
+import { TinyHooks, TinySelections, TinyContentActions, McEditor, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import QuickbarsPlugin from 'tinymce/plugins/quickbars/Plugin';
+
+describe('browser.tinymce.core.focus.KeyboardShortcutContextualMenuTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'quickbars',
+    quickbars_insert_toolbar: 'quicktable image media codesample',
+    quickbars_selection_toolbar: 'bold italic underline | formatselect | bullist numlist | blockquote quicklink',
+    contextmenu: 'undo redo | inserttable | cell row column deletetable | help',
+    base_url: '/project/tinymce/js/tinymce',
+  }, [ QuickbarsPlugin ]);
+
+  const doc = SugarDocument.getDocument();
+
+  it('TINY-2884: keypress should focus on contextual menu', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>aaaaaaa</p>');
+    editor.focus();
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 4);
+    setTimeout(() => {
+      TinyContentActions.keystroke(editor, 120, { ctrl: true });
+    }, 500);
```

Comment:
Why does this need to be in a `setTimeout`? Also if it does need to be in a timeout then why does this have a 500ms delay? That's far too long for tests generally and we normally aim for 50ms or less if possible.

Reply:
could just be my machine is buggy at times, this has been removed https://github.com/tinymce/tinymce/pull/7533/commits/6f20447b960bd398b43e96efc00dbad2dd656b76